### PR TITLE
finishing module.go and adding client query/tx

### PR DESCRIPTION
**Pull request by adisaran64**
Finished modifying the module.go file (sets up the `AppModule` interfaces, and binds ports) and added the cli/client/tx.go and cli/client/query.go files (to set up the client).

The cli/client/query.go file still has to be modified further after adding the msg-server.go file to support doing the IBC-EVM call from the command line, the skeleton is there for now to prevent errors in the module.go file.

Closes: #XXX

______

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.

PR review checkboxes:

I have...

- [ ] added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/evmos/evmos/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link in the PR description to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all required CI checks have passed

Code maintenance:

I have...

- [ ] written unit and integration [tests](https://github.com/evmos/evmos/blob/main/CONTRIBUTING.md#testing)
- [ ] added relevant [`godoc`](https://go.dev/blog/godoc) and [code comments](https://blog.jbowen.dev/2019/09/the-magic-of-go-comments/).
- [ ] updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)

______

### Reviewers Checklist

**All** items are required. Please add a note if the item is not applicable and please add your handle next to the items reviewed if you only reviewed selected items.

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] confirmed that this PR does not change production code

### CODE REVIEW

**Comment by adisaran64** [comment]
TODO: need to add to the client/cli/query.tx file after finishing the msgServer

**Comment by danburck** [comment]
@adisaran64 Can you add a description please? I'm missing context

**Comment by adisaran64** [comment]
@danburck accidentally left the TODO comment, meant to just leave it in the code in the `client/cli/tx.go` file. added a description.

File: x/ibc/evm/client/cli/query.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	return cmd
+}
+
+// GetParamsCmd queries ibc-evm-tx module params
+func GetParamsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "params",
+		Short: "Gets ibc-evm-tx params",
+		Long:  "Gets ibc-evm-tx params",
```
Comment (GAtom22) [inline x/ibc/evm/client/cli/query.go:37]: We renamed the module to IBC EVM to be more generic. So in the future we could do queries also for example
```suggestion
// GetParamsCmd queries ibc-evm module params
func GetParamsCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "params",
		Short: "Gets ibc-evm params",
		Long:  "Gets ibc-evm params",

```

File: x/ibc/evm/client/cli/tx.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+
+	"github.com/evmos/evmos/v10/x/ibc/evm/types"
+)
+
+// NewTxCmd returns a root CLI command handler for ibc-evm-tx transaction commands
+func NewTxCmd() *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "ibc-evm-tx subcommands",
```
Comment (GAtom22) [inline x/ibc/evm/client/cli/tx.go:15]:
```suggestion
// NewTxCmd returns a root CLI command handler for ibc-evm transaction commands
func NewTxCmd() *cobra.Command {
	txCmd := &cobra.Command{
		Use:                        types.ModuleName,
		Short:                      "ibc-evm subcommands",

```

File: x/ibc/evm/module.go [hunk 3]
```diff
@@ -80,21 +80,23 @@ func (AppModuleBasic) GetQueryCmd() *cobra.Command {
...
 type AppModule struct {
 	AppModuleBasic
-	keeper keeper.Keeper
+	keeper    keeper.Keeper
+	msgServer types.MsgServer
 }
 
-// NewAppModule creates a new 20-transfer module
-func NewAppModule(k keeper.Keeper) AppModule {
+// NewAppModule creates a new ibc-evm-tx module
```
Comment (GAtom22) [inline x/ibc/evm/module.go:87]:
```suggestion
// NewAppModule creates a new ibc-evm module
```

File: x/ibc/evm/module.go [hunk 4]
```diff
@@ -119,18 +121,17 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
...
-	// var genesisState types.GenesisState
-	// cdc.MustUnmarshalJSON(data, &genesisState)
-	// am.keeper.InitGenesis(ctx, genesisState)
+	var genesisState types.GenesisState
+	cdc.MustUnmarshalJSON(data, &genesisState)
+	am.keeper.InitGenesis(ctx, genesisState)
 	return []abci.ValidatorUpdate{}
 }
 
 // ExportGenesis returns the exported genesis state as raw bytes for the ibc-transfer
```
Comment (GAtom22) [inline x/ibc/evm/module.go:130]:
```suggestion
// ExportGenesis returns the exported genesis state as raw bytes for the ibc-evm
```

File: x/ibc/evm/module.go [hunk 5]
```diff
@@ -119,18 +121,17 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 // InitGenesis performs genesis initialization for the ibc-transfer module. It returns
```
Comment (GAtom22) [inline x/ibc/evm/module.go:121]:
```suggestion
// InitGenesis performs genesis initialization for the ibc-evm module. It returns
```

File: x/ibc/evm/types/genesis.go [hunk 6]
```diff
@@ -1,15 +1,26 @@
...
+
 // NewGenesisState creates a new ibc-transfer GenesisState instance.
 func NewGenesisState(params Params) *GenesisState {
 	return &GenesisState{
+		PortId: PortID,
 		Params: params,
 	}
 }
 
 // DefaultGenesisState returns a GenesisState with "transfer" as the default PortID.
```
Comment (GAtom22) [inline x/ibc/evm/types/genesis.go:13]:
```suggestion
// DefaultGenesisState returns a GenesisState with "ibc-evm" as the default PortID.
```

**Review by GAtom22 — CHANGES_REQUESTED** [review]
LGTM! Added a few comments regarding the module name. We renamed to `ibc-evm` instead of `ibc-evm-tx` to be more generic

File: x/ibc/evm/client/cli/query.go (hunk 1 above)
Comment (danburck) [reply x/ibc/evm/client/cli/query.go:37]: could also use 

```
fmt.Sprintf("Gets %s  params", types.ModuleName)
```

File: x/ibc/evm/client/cli/query.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	)
+
+	return cmd
+}
+
+// GetParamsCmd queries ibc-evm module params
+func GetParamsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "params",
+		Short: "Gets ibc-evm params",
```
Comment (danburck) [inline x/ibc/evm/client/cli/query.go:36]:
```suggestion
		Short: fmt.Sprintf("Gets %s params", types.ModuleName),

```

File: x/ibc/evm/client/cli/tx.go [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+
+	"github.com/evmos/evmos/v10/x/ibc/evm/types"
+)
+
+// NewTxCmd returns a root CLI command handler for ibc-evm transaction commands
+func NewTxCmd() *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "ibc-evm subcommands",
```
Comment (danburck) [inline x/ibc/evm/client/cli/tx.go:15]: use sprintf here too

File: x/ibc/evm/keeper/genesis.go [hunk 9]
```diff
@@ -6,18 +6,18 @@ import (
 	"github.com/evmos/evmos/v10/x/ibc/evm/types"
 )
 
-// InitGenesis import module genesis
-func InitGenesis(
+// InitGenesis sets module genesis
```
Comment (danburck) [inline x/ibc/evm/keeper/genesis.go:9]: Move `genesis.go` file to `x/ibc/evm/genesis.go`

File: x/ibc/evm/module.go [hunk 10]
```diff
@@ -80,21 +80,23 @@ func (AppModuleBasic) GetQueryCmd() *cobra.Command {
...
+		keeper:    k,
+		msgServer: ms,
 	}
 }
 
 // RegisterInvariants implements the AppModule interface
 func (AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {}
 
 func (am AppModule) NewHandler() sdk.Handler {
-	return NewHandler(am.keeper)
```
Comment (danburck) [inline x/ibc/evm/module.go:97]: You can pass the am.keeper here, there is no need to add a msgServer in `AppModule`

**Pull request closed without merging**